### DISCUSSION TITLE
MPP on AMD Epyc processor

### DISCUSSION THREAD

**Discussion by anghingo** [OpenRadioss/OpenRadioss#3211] [Users]
Hi,
I’m an old LLNL dyna3d and ls-dyna user (30+ years) and now I’m moving together with all my staff to Open Radioss that I believe is the future. We installed the latest version including Intel one API on several Intel based servers and everything works fine (we are learning ..) but on a server based on AMD Epyc processor the MPP and HMPP versions do not run. Is there a dedicated tool or procedure for AMD processors (instead of Intel oneAPI)?
Thanks
Marco

**Comment by OlivierWienholtz** [comment]
Hello,
We run successfully OpenRadioss on AMD Epyc Servers (Linux & Windows).
Intel MPI works on AMD servers.
We have seen Affinity issues on Windows, setting the variable 
KMP_AFFINITY=disabled 
helps.

What is the Operating system on the Epyc Server ? 
Can you give us more insights about your problems ? 

Best Regards,
Olivier

**Comment by anghingo** [comment]
Hello Olivier,

Thanks for the prompt answer.

The sever is a Dell with:
Windows Server 2022 Standard
AMD EPYC 7H12 64-Core Processor  2.60 GHz  (2 processors)
256 GB ram
Visual Studio Build Tools installed
Intel oneAPI installed and setvars.bat has been run

To manage Open Radioss (version 20241114) I use:
OpenRadioss_gui.vbs
***@***.***

As input file I’m using (for all the servers) a vehicle impact. Ls-dyna file (attached). In all the Intel severs the model run (with -np = numer of cores) with equivalent performances.
In the AMD server:
If I specify -nt (example 64) the simulation starts but is very slow.
If I specify -np (any number) after the input file verification the solver does not start…
***@***.***

Remains with the above message.

Thanks

Marco

Prof. Marco Anghileri
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano
via La Masa 34
Ufficio: laboratorio Last
Edificio 6 scala C
Via Candiani 72, Milano
tel +39 02 2399  7162 - 7152
fax +39 02 2399 7153
E-mail ***@***.******@***.***>

Da: Olivier Wienholtz ***@***.******@***.***>>
Inviato: giovedì 12 dicembre 2024 10:15
A: OpenRadioss/OpenRadioss ***@***.******@***.***>>
Cc: Marco Anghileri ***@***.******@***.***>>; Author ***@***.******@***.***>>
Oggetto: Re: [OpenRadioss/OpenRadioss] MPP on AMD Epyc processor (Discussion #3211)


Hello,
We run successfully OpenRadioss on AMD Epyc Servers (Linux & Windows).
Intel MPI works on AMD servers.
We have seen Affinity issues on Windows, setting the variable
KMP_AFFINITY=disabled
helps.

What is the Operating system on the Epyc Server ?
Can you give us more insights about your problems ?

Best Regards,
Olivier

**Comment by OlivierWienholtz** [comment]
Hello,

Here are some possible reasons:

The GUI does not load IntelMPI as it does not know where to find the distribution. Not only the Path to MPI is need but other variables.
There is an issue logged for that : https://github.com/OpenRadioss/Tools/issues/25
Pls call the IntelMPI environment : [Path to IntelMPI]\env\vars.bat in a shell before launching the in the same shell:
In typical installation in cmd.exe
call C:\Program Files (x86)\Intel\oneAPI\mpi\latest\env\vars.bat
cd [Path to OpenRadioss]\openradioss_gui
openradioss_gui.bat

If this does not work, pls send error messages, screenshots etc...

Thank You,
Best Regards,
Olivier

**Comment by bernoertel** [comment]
Hello Prof. Anghileri, how do you come to the conclusion, that Open Radioss is the future. Can you short explain  ?

**Comment by anghingo** [comment]
Because is restarting the old dyna-3d approach. A software where you can add and work on the parts you are interested.

Marco Anghileri
Politecnico di Milano
Dipartimento di Ingegneria Aerospaziale
Inviato da iPhone

Il giorno 2 gen 2025, alle ore 10:54, bernoertel ***@***.***> ha scritto:

﻿

Hello Prof. Anghileri, how do you come to the conclusion, that Open Radioss is the future. Can you short explain ?
